### PR TITLE
v2 IGP interface change

### PR DESCRIPTION
**Pull request by tkporter**
### Description

* Moves to the new IGP interface without changing any functionality.
* Renames `payGasFor` -> `payForGas`
* For HelloWorld, decided to act like the gas amount passed in will actually be treated as the amount used for the `handle` function. Even though this isn't yet the case, it prepares for the future so we don't need to deploy again and provides an example for devs

### Drive-by changes

* Had some Typescript issues where in `utils` we were trying to import `@hyperlane-xyz/utils` - instead just moved to importing them from a relative path
* Fixed the `update_abis.sh` script to work for v2
* Renamed the `amount` field on a rust `InterchainGasPayment` to `payment` to avoid confusion with the gas amount
* Changed the Router's `_dispatchWithGas` function to always call the IGP even if the msg.value is 0. This is so that we'll be able to get the behavior we want with on-chain fee quoting-- which is to revert if the IGP says that not enough payment has been received. If we kept that if statement, txs calling `_dispatchWithGas` wouldn't revert if they paid 0 `msg.value` even if the IGP had implemented on chain fee quoting!

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1296

### Backward compatibility

_Are these changes backward compatible?_

No - IGP has an interface change!

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

None

### Testing

_What kind of testing have these changes undergone?_

Unit Tests, ran `run-locally`

### CODE REVIEW

File: solidity/test/router.test.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ const destination = 2;
 const destinationWithoutRouter = 3;
 const body = '0xdeadbeef';
 
+interface GasPaymentParams {
+  gasAmount: BigNumberish;
+  gasPayment: BigNumberish;
```
Comment (yorhodes) [inline solidity/test/router.test.ts:29]: do we need the `gas` prefix? not immediately clear what's different between `gasAmount` and `gasPayment`

File: typescript/helloworld/contracts/HelloWorld.sol [hunk 2]
```diff
@@ -9,6 +9,11 @@ import {Router} from "@hyperlane-xyz/core/contracts/Router.sol";
  * @dev You can use this simple app as a starting point for your own application.
  */
 contract HelloWorld is Router {
+    /// A generous upper bound on the amount of gas to use in the handle function
+    /// when a message is processed. Consider moving to a mapping to accommodate
+    /// other execution environments.
+    uint256 public constant HANDLE_GAS_AMOUNT = 100_000;
```
Comment (yorhodes) [inline typescript/helloworld/contracts/HelloWorld.sol:15]: I see low harm in making this modifiable or a mapping now

File: solidity/contracts/InterchainGasPaymaster.sol [hunk 3]
```diff
@@ -17,9 +17,10 @@ contract InterchainGasPaymaster is IInterchainGasPaymaster, OwnableUpgradeable {
     /**
      * @notice Emitted when a payment is made for a message's gas costs.
      * @param messageId The ID of the message to pay for.
-     * @param amount The amount of native tokens paid.
+     * @param payment The amount of native tokens paid.
+     * @param gas The amount of destination gas paid for.
```
Comment (yorhodes) [inline solidity/contracts/InterchainGasPaymaster.sol:21]: these could be named more clearly

**Review by yorhodes — APPROVED** [review]

**Comment by asaj** [comment]
Changed the base to `main`

**Review by yorhodes — APPROVED** [review]

**Pull request merged**